ksqldb-api-client/src/test/java/io/confluent/ksql/api/client/ClientTest.java

```diff
@@ -168,10 +223,42 @@ public void shouldHandleErrorResponseFromStreamQuery() {
     );
 
     // Then
+    assertThat(e.getCause(), instanceOf(KsqlRestClientException.class));
     assertThat(e.getCause().getMessage(), containsString("Received 400 response from server"));
     assertThat(e.getCause().getMessage(), containsString("invalid query blah"));
   }
 
+  @Test
+  public void shouldFailPollStreamedQueryResultIfSubscribed() throws Exception {
+    // Given
+    final StreamedQueryResult streamedQueryResult =
+        javaClient.streamQuery(DEFAULT_PUSH_QUERY, DEFAULT_PUSH_QUERY_REQUEST_PROPERTIES).get();
+    streamedQueryResult.subscribe(new TestSubscriber<>());
+
+    // When
+    final Exception e = assertThrows(IllegalStateException.class, streamedQueryResult::poll);
+
+    // Then
+    assertThat(e.getMessage(), containsString("Cannot poll if subscriber has been set"));
+  }
+
+  @Test
+  public void shouldFailSubscribeStreamedQueryResultIfPolling() throws Exception {
+    // Given
+    final StreamedQueryResult streamedQueryResult =
+        javaClient.streamQuery(DEFAULT_PUSH_QUERY, DEFAULT_PUSH_QUERY_REQUEST_PROPERTIES).get();
+    streamedQueryResult.poll(1, TimeUnit.NANOSECONDS);
+
+    // When
+    final Exception e = assertThrows(
+        IllegalStateException.class,
+        () -> streamedQueryResult.subscribe(new TestSubscriber<>())
+    );
+
+    // Then
+    assertThat(e.getMessage(), containsString("Cannot set subscriber if polling"));
+  }
+
```

Comment:
Would be nice to add a couple of tests which do the following:
* Terminate the connection from the server side and verify that the query results is completed on the client
* Ends the response (before limit) on the server side and verify that the query result is complete on client.

Reply:
What do you mean by "terminate the connection from the server"? Would that be if the server encounters an exception, or something else? Is this the same as "ends the response on the server side"?

Reply:
I guess the connection could also be terminated if the query were terminated via `/close-query`. I can add a test for that as well.